ros2bag_tools/ros2bag_tools/filter/restamp.py

```diff
@@ -72,10 +73,21 @@ def filter_msg(self, serialized_msg):
         if hasattr(msg, 'header'):
             header_time = Time.from_msg(msg.header.stamp)
             t = header_time.nanoseconds
+        elif isinstance(msg, TFMessage):
+            times = [Time.from_msg(transform.header.stamp).nanoseconds for transform in msg.transforms if hasattr(transform, 'header')]
+            if len(times) > 0:
+                t = min(times)
         if topic in self._offset_topics:
             t = t + self._offset.nanoseconds
             if hasattr(msg, 'header') and self._modify_header:
                 header_time += self._offset
                 msg.header.stamp = header_time.to_msg()
                 data = serialize_message(msg)
+            elif isinstance(msg, TFMessage) and self._modify_header:
+                for transform in msg.transforms:
+                    if hasattr(transform, 'header'):
+                        header_time = Time.from_msg(transform.header.stamp)
+                        header_time += self._offset
+                        transform.header.stamp = header_time.to_msg()
+                        data = serialize_message(msg)
```

Comment:
serialize_message() should only be called per-message, not per-transform like here